File: src/Entity/Equipement.php
<?php

namespace App\Entity;

use App\Entity\Categorie2eq;
use App\Repository\EquipementRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;


/**
 * @ORM\Entity(repositoryClass=EquipementRepository::class)
 */
class Equipement
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("post:read")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     * @Groups("post:read")
     */
    private $name;


    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     * @Groups("post:read")
     */
    private $marque;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank
     * @Groups("post:read")
     */
    private $description;

    /**
     * @ORM\ManyToOne(targetEntity=categorie2eq::class, inversedBy="equipements")
     * @ORM\JoinColumn(nullable=false)
     */
    private $categorie2eq;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups("post:read")
     */
    private $image;

    /**
     * @ORM\Column(type="float")
     * @Groups("post:read")
     */
    private $prix;


    /**
     * @ORM\ManyToMany(targetEntity=Commande::class, mappedBy="listP")
     */
    private $commandes;





    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): self
    {
        $this->name = $name;

        return $this;
    }


    public function getMarque(): ?string
    {
        return $this->marque;
    }

    public function setMarque(string $marque): self
    {
        $this->marque = $marque;

        return $this;
    }


    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getCategorie2eq(): ?Categorie2eq
    {
        return $this->categorie2eq;
    }

    public function setCategorie2eq(?Categorie2eq $categorie2eq): self
    {
        $this->categorie2eq = $categorie2eq;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    /**
     * @return Collection|EquipementLike[]
     */
    public function getLikes(): Collection
    {
        return $this->likes;
    }

    public function addLike(EquipementLike $like): self
    {
        if (!$this->likes->contains($like)) {
            $this->likes[] = $like;
            $like->setEquipement($this);
        }

        return $this;
    }

    public function removeLike(EquipementLike $like): self
    {
        if ($this->likes->removeElement($like)) {
            // set the owning side to null (unless already changed)
            if ($like->getEquipement() === $this) {
                $like->setEquipement(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection<int, Commande>
     */
    public function getCommandes(): Collection
    {
        return $this->commandes;
    }

    public function addCommande(Commande $commande): self
    {
        if (!$this->commandes->contains($commande)) {
            $this->commandes[] = $commande;
            $commande->addListP($this);
        }

        return $this;
    }

    public function removeCommande(Commande $commande): self
    {
        if ($this->commandes->removeElement($commande)) {
            $commande->removeListP($this);
        }

        return $this;
    }

}

File: src/entities/Equipement.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

/**
 *
 * @author wided
 */
public class Equipement {
    private int id,prix;
    private String name,marque,description;

    public Equipement() {
    }

    public Equipement(int id, String name, String marque, String description, int prix) {
        this.id = id;
        this.name = name;
        this.marque = marque;
        this.description = description;
        this.prix = prix;
    }

    public Equipement(int id, int prix) {
        this.id = id;
        this.prix = prix;
    }

    public Equipement(int prix) {
        this.prix = prix;
    }

    public Equipement(int prix, String name, String marque, String description) {
        this.prix = prix;
        this.name = name;
        this.marque = marque;
        this.description = description;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPrix() {
        return prix;
    }

    public void setPrix(int prix) {
        this.prix = prix;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getMarque() {
        return marque;
    }

    public void setMarque(String marque) {
        this.marque = marque;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Equipement{" + "id=" + id + ", prix=" + prix + ", name=" + name + ", marque=" + marque + ", description=" + description + '}';
    }
    
    
}
